boards/arm/wio_terminal/wio_terminal.dts

```diff
@@ -0,0 +1,266 @@
+		uart0 = &sercom2;
+		accel0 = &lis3dh;
+		sdhc0 = &sdhc0;
+	};
+
+	/* LEDs */
+	leds {
+		compatible = "gpio-leds";
+		led0: led_0 {
+			label = "LED";
+			gpios = <&porta 15 GPIO_ACTIVE_HIGH>;
+		};
+	};
+
+	/* Buttons */
+	gpio_keys {
+		compatible = "gpio-keys";
+		user_button_0: button_0 {
+			label = "User Button 0";
+			gpios = <&portc 26 GPIO_ACTIVE_LOW>;
+		};
+		user_button_1: button_1 {
+			label = "User Button 1";
+			gpios = <&portc 27 GPIO_ACTIVE_LOW>;
+		};
+		user_button_2: button_2 {
+			label = "User Button 2";
+			gpios = <&portc 28 GPIO_ACTIVE_LOW>;
+		};
+		joy_sel: joystick_selection {
+			label = "joystick selection";
+			gpios = <&portd 10 GPIO_ACTIVE_LOW>;
+		};
+		joy_down: joystick_down {
+			label = "joystick down";
+			gpios = <&portd 8 GPIO_ACTIVE_LOW>;
+		};
+		joy_up: joystick_up {
+			label = "joystick up";
+			gpios = <&portd 20 GPIO_ACTIVE_LOW>;
+		};
+		joy_left: joystick_left {
+			label = "joystick left";
+			gpios = <&portd 12 GPIO_ACTIVE_LOW>;
+		};
+		joy_right: joystick_right {
+			label = "joystick right";
+			gpios = <&portd 9 GPIO_ACTIVE_LOW>;
+		};
+	};
+
+	/* Regulators */
+	lcd_backlight_en {
+		compatible = "regulator-fixed";
+		regulator-name = "lcd_backlight_enable";
+		enable-gpios = <&portc 5 GPIO_ACTIVE_HIGH>;
+		regulator-boot-on;
+	};
+	rpi_power_3v3_en {
+		compatible = "regulator-fixed";
+		regulator-name = "rpi_power_3v3_enable";
+		enable-gpios = <&portc 15 GPIO_ACTIVE_LOW>;
+		regulator-boot-on;
+	};
+	rpi_power_5v_en {
+		compatible = "regulator-fixed";
+		regulator-name = "rpi_power_5v_enable";
+		enable-gpios = <&portc 14 GPIO_ACTIVE_HIGH>;
+		regulator-boot-on;
+	};
+	usb_power_5v_en {
+		compatible = "regulator-fixed";
+		regulator-name = "usb_power_5v_en";
+		enable-gpios = <&porta 27 GPIO_ACTIVE_LOW>;
+	};
+};
+
+&cpu0 {
+	clock-frequency = <120000000>;
+};
+
+&dmac {
+	status = "okay";
+};
+
+&flash0 {
+	partitions {
+		compatible = "fixed-partitions";
+		#address-cells = <1>;
+		#size-cells = <1>;
+
+		boot_partition: partition@0 {
+			label = "uf2";
+			reg = <0x00000000 DT_SIZE_K(16)>;
+			read-only;
+		};
+
+		code_partition: partition@4000 {
+			label = "code";
+			reg = <0x4000 DT_SIZE_K(512-16)>;
+			read-only;
+		};
+	};
+};
+
+&sercom0 {
+	status = "okay";
+	compatible = "atmel,sam0-spi";
+	#address-cells = <1>;
+	#size-cells = <0>;
+	dipo = <2>;
+	dopo = <0>;
+	pinctrl-0 = <&sercom0_spi_default>;
+	pinctrl-names = "default";
+	cs-gpios = <&portc 25 GPIO_ACTIVE_LOW>;
+};
+
+&sercom1 {
+	status = "okay";
+	compatible = "atmel,sam0-uart";
+	current-speed = <115200>;
+	rxpo = <1>;
+	txpo = <0>;
+	pinctrl-0 = <&sercom1_uart_default>;
+	pinctrl-names = "default";
+};
+
+&sercom2 {
+	status = "okay";
+	compatible = "atmel,sam0-uart";
+	current-speed = <115200>;
+	rxpo = <1>;
+	txpo = <0>;
+	pinctrl-0 = <&sercom2_uart_default>;
+	pinctrl-names = "default";
+};
+
+&sercom3 {
+	status = "okay";
+	compatible = "atmel,sam0-i2c";
+	clock-frequency = <I2C_BITRATE_FAST>;
```

Comment:
should this be `I2C_BITRATE_HIGH`?

Reply:
There is no strong solution. To be honest I'm in favor of the lowest frequency (100kbits/s), but it's not working due to https://github.com/zephyrproject-rtos/zephyr/issues/42432. So I have set 400kbits/s.
By the way, the user making a real project can configure higher clock if required, making an overlay.
But is it required for a small accelerometer or other Groove sensors ? For most projects I think the answer is no.